openupgrade_scripts/scripts/mail/17.0.1.15/pre-migration.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_models_renames = [
+    ("mail.channel", "discuss.channel"),
+    ("mail.channel.member", "discuss.channel.member"),
+    ("mail.channel.rtc.session", "discuss.channel.rtc.session"),
+]
+_tables_renames = [
+    ("mail_channel", "discuss_channel"),
+    ("mail_channel_member", "discuss_channel_member"),
+    ("mail_channel_rtc_session", "discuss_channel_rtc_session"),
+]
+_fields_renames = [
+    (
+        "discuss.channel",
+        "discuss_channel",
+        "field",
+        "field_id",
+    ),
+]
+
+
+def _discuss_channel_fill_allow_public_upload(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE discuss_channel
+        ADD COLUMN IF NOT EXISTS allow_public_upload BOOLEAN;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE discuss_channel
+        SET allow_public_upload = True
+        WHERE channel_type = 'livechat'
+        """,
+    )
+
+
+def _mail_alias_fill_multiple_values(env):
+    """
+    We will fill value for alias_full_name in post because alias_domain has not been
+    present yet
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mail_alias
+        ADD COLUMN IF NOT EXISTS alias_full_name VARCHAR,
+        ADD COLUMN IF NOT EXISTS alias_incoming_local BOOLEAN,
+        ADD COLUMN IF NOT EXISTS alias_status VARCHAR;
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_alias
+        SET alias_incoming_local = True
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE mail_alias
+        SET alias_status = 'not_tested'
```

Comment:
why not one query?

Further, I think we should assume that we get a correctly configured database and we migrate correct configurations correctly, so the alias status should be `valid` in my opinion.